### PR TITLE
[JSON-API] Add list users endpoint

### PR DESCRIPTION
**Pull request by realvictorprm**
changelog_begin

- [HTTP-JSON] Added an endpoint /users which returns the available users on the ledger.

changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala [hunk 1]
```diff
@@ -156,6 +156,11 @@ object domain extends com.daml.fetchcontracts.domain.Aliases {
 
   final case class UserDetails(userId: String, primaryParty: Option[String])
 
+  object UserDetails {
+    def fromUser(user: User) =
+      UserDetails(user.id, user.primaryParty.map(_.toString))
```
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala:161]: Don't call toString on Strings.

```suggestion
      UserDetails(user.id, user.primaryParty)
```

**Review by S11001001 — APPROVED** [review]
Looks good, thanks.

**Comment by S11001001** [comment]
By the way, this does still need documentation.

**Comment by realvictorprm** [comment]
> By the way, this does still need documentation.

Will add documentation for the new endpoints in a separate PR

**Pull request merged**